README.md

```diff
@@ -164,115 +105,13 @@ There are three key concepts in Spectral: **Rulesets**, **Rules** and **Function
 
-#### Creating a custom rule
-
-Spectral has a built-in set of functions which you can reference in your rules. This example uses the `RuleFunction.PATTERN` to create a rule that checks that all property values are in snake case.
-
-```javascript
-const { RuleFunction, Spectral } = require('@stoplight/spectral');
-
-const spectral = new Spectral();
-
-spectral.addRules({
-  snake_case: {
-    summary: 'Checks for snake case pattern',
-
-    // evaluate every property
-    given: '$..*',
-
-    then: {
-      function: RuleFunction.PATTERN,
-      functionOptions: {
-        match: '^[a-z]+[a-z0-9_]*[a-z0-9]+$',
-      },
-    },
-  },
-});
-
-// run!
-spectral.run({name: 'helloWorld',}).then(results => {
-  console.log(JSON.stringify(results, null, 4));
-});
-
-// => outputs a single result since `helloWorld` is not snake_case
-// [
-//   {
-//     "code": "snake_case",
-//     "message": "must match the pattern '^[a-z]+[a-z0-9_]*[a-z0-9]+$'",
-//     "severity": 1,
-//     "path": [
-//       "name"
-//     ]
-//   }
-// ]
-```
+## Programmatic usage
 
-#### Creating a custom function
-
-Sometimes the built-in functions don't cover your use case. This example creates a custom function, `customNotThatFunction`, and then uses it within a rule, `openapi_not_swagger`. The custom function checks that you are not using a specific string (e.g., "Swagger") and suggests what to use instead (e.g., "OpenAPI").
-
-```javascript
-const { Spectral } = require('@stoplight/spectral');
-
-// custom function
-const customNotThatFunction = (targetValue, options) => {
-  const { match, suggestion } = options;
-
-  if (targetValue && targetValue.match(new RegExp(match))) {
-    // return the single error
-    return [
-      {
-        message: `Use ${suggestion} instead of ${match}!`,
-      },
-    ];
-  }
-};
-
-const spectral = new Spectral();
-
-spectral.addFunctions({
-  notThat: customNotThatFunction,
-});
-
-spectral.addRules({
-  openapi_not_swagger: {
-    summary: 'Checks for use of Swagger, and suggests OpenAPI.',
-
-    // check every property
-    given: '$..*',
-
-    then: {
-      // reference the function we added!
-      function: 'notThat',
-
-      // pass it the options it needs
-      functionOptions: {
-        match: 'Swagger',
-        suggestion: 'OpenAPI',
-      },
-    },
-  },
-});
-
-// run!
-spectral.run({description: 'Swagger is pretty cool!',}).then(results => {
-  console.log(JSON.stringify(results, null, 4));
-});
-
-// => outputs a single result since we are using the term `Swagger` in our object
-// [
-//   {
-//     "code": "openapi_not_swagger",
-//     "message": "Use OpenAPI instead of Swagger!",
-//     "severity": 1,
-//     "path": [
-//       "description"
-//     ]
-//   }
-// ]
-```
+Spectral is written in TypeScript (JavaScript) and can be used directly for when working with the CLI isn't the right approach. Take a look at our ["JavaScript API documentation"](docs/js-api.md).
```

Comment:
Instead of saying "right approach," say "for when you need to use Spectral programatically."